tests/test_lua_consumer.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+
+from comdb2 import dbapi2
+from comdb2 import factories
+
+MONITOR_PROCEDURE = """
+local function main()
+    db:column_type('int', 1)
+    db:column_name('id', 1)
+
+    db:column_type('text', 2)
+    db:column_name('old_message', 2)
+
+    db:column_type('text', 3)
+    db:column_name('new_message', 3)
+
+    local consumer = db:consumer()
+    while true do
+        local change = consumer:get()
+        local row = {id=nil, old_message=nil, new_message=nil}
+        if change.new ~= nil then
+            row.id = change.new.id
+            row.new_message = change.new.message
+        end
+        if change.old ~= nil then
+            row.id = change.old.id
+            row.old_message = change.old.message
+        end
+
+        consumer:emit(row)
+        consumer:consume()
+    end
+end
+"""
+
+
+@pytest.fixture(autouse=True)
```

Comment:
not sure what the default scope is (wasn't it function?) but it might be good to explicitly set module here.

Reply:
The scope on this should be what we want - it will be used for each function in this module, but won't be used for functions defined in other test modules. I think. I'll double check when I get a chance.

Reply:
Yeah, this does what I wanted it to: it runs once for each test function in this test module, and not at all for test functions in other test modules.